src/service.ts

```diff
@@ -6,6 +6,42 @@ import * as k8s from './imports/k8s';
 import * as ingress from './ingress';
 import * as pod from './pod';
 
+/**
+ * Interface for Service constructs.
+ * This interface allows users to interact with a Service without having to create one directly.
+ */
+export interface IService extends base.IResource {
+  /**
+   * The ports this service binds to.
+   */
+  readonly ports: ServicePort[];
+
+  /**
+   * The first port of the service, if available.
+   */
+  readonly port?: number;
+
+  /**
+   * Expose a service via an ingress using the specified path.
+   *
+   * @param path The path to expose the service under.
+   * @param options Additional options.
+   *
+   * @returns The `Ingress` resource that was used.
+   */
+  exposeViaIngress(path: string, options?: ExposeServiceViaIngressOptions): ingress.Ingress;
+
+  /**
+   * Require this service to select pods matching the selector.
+   *
+   * Note that invoking this method multiple times acts as an AND operator
+   * on the resulting labels.
+   *
+   * @param selector The pod selector
+   */
+  select(selector: pod.IPodSelector): void;
```

Comment:
Why did you add this to the interface if it cannot be implemented by imported services?